draft/0005-impersonation.md

```diff
@@ -6,31 +6,32 @@ Revision: 1
+## 2. Motivation
 
 As an enterprise-ready identity provider, it is essential to provide user impersonation capabilities to support various use cases. For example, a support engineer may need to troubleshoot an issue on behalf of a user, or an administrator may need to verify the permissions of a user. User impersonation can be helpful to these processes and improve the overall user experience.
 
-## Token exchange
+## 3. Token exchange
 
 The user impersonation feature will be implemented using the OAuth 2.0 [token exchange flow](https://www.rfc-editor.org/rfc/rfc8693.html). The token exchange flow allows an authorized user to exchange their access token for a new access token that represents the identity of another user. The new access token can then be used to make API requests on behalf of the impersonated user.
 
-### Request
+### 3.1 Request
 
 The client makes a token exchange request to the tenant's token endpoint with a special grant type using the HTTP POST method. The following parameters are included in the HTTP request entity-body using the `application/x-www-form-urlencoded` format.
 
 1. `grant_type`: REQUIRED. The value of this parameter must be `urn:ietf:params:oauth:grant-type:token-exchange` indicates that a token exchange is being performed.
-2. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
-3. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
-4. `subject_token`: REQUIRED. A security token that represents the identity of the party on behalf of whom the request is being made. See the next section for more details.
-5. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
-6. `actor_token`: OPTIONAL. A security token that represents the identity of the acting party, it should contain valid `accountId` and have `openid` scope. If present, the access token issued by Logto will contain an additional `act` claim that represents the identity of the acting party.
-7. `actor_token_type`: OPTIONAL. The type of the security token provided in the `actor_token` parameter. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
+2. `client_id`: REQUIRED. The client identifier of the client application that is making the request, the returned access token will contain this `client_id` claim.
+3. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+4. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+5. `subject_token`: REQUIRED. A security token that represents the identity of the party on behalf of whom the request is being made. See the next section for more details.
+6. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
```

Comment:
nit: should we reorder the parameters and put REQUIRED params before OPTIONAL params?

Reply:
The first 4 params are common in other grants, and then comes the new params.